### PR TITLE
[KEYCLOAK-8449] - Option to automatically map HTTP verbs to scopes when configuring the policy enforcer

### PR DESCRIPTION
**Pull request by pedroigor**
Documentation: https://github.com/keycloak/keycloak-documentation/pull/494

### CODE REVIEW

**Review by oridox — DISMISSED** [review]

File: adapters/oidc/adapter-core/src/main/java/org/keycloak/adapters/authorization/AbstractPolicyEnforcer.java
Comment (dbayub) [inline adapters/oidc/adapter-core/src/main/java/org/keycloak/adapters/authorization/AbstractPolicyEnforcer.java]: Since this still loads all of the scopes associated with the resource as defined in the server, you would never get an empty scopes list. You have to associate the scopes with the resource in order to be able to define a scope based permission (unless something else has changed.) So I think that the code should be:

List<String> scopes = new ArrayList<>();

if (Boolean.TRUE.equals(getEnforcerConfig().getHttpMethodAsScope())) {
    scopes.add(request.getMethod());
} else {
    scopes.addAll(pathConfig.getScopes());
}

Comment (pedroigor) [reply adapters/oidc/adapter-core/src/main/java/org/keycloak/adapters/authorization/AbstractPolicyEnforcer.java]: You are right. I've fixed and also updated test to make sure we behave as expected when resources are associated with scopes.

**Review by mhajas — APPROVED** [review]

**Review by stianst — APPROVED** [review]
Approving without review, assuming @mhajas has reviewed this

**Comment by stianst** [comment]
@pedroigor please wait for Jenkins CI to complete before merging PRs

**Pull request merged**
